storage/mongodb/mongodb-test.yaml

```diff
@@ -0,0 +1,149 @@
+        fail:
+          msg: "the pod is NOT deleted after 120s"
+        when: get_project_result.rc == 0
+    when: delete_test_project_before_test|bool == true
+    tags:
+      - setup
+
+  - name: create test project
+    command:  oc new-project {{ test_project_name }}
+    tags:
+      - setup
+
+  - name: create mongodb
+    shell: oc process -f {{ tmp_folder }}/files/oc/mongodb-persistent-template.yaml -p MEMORY_LIMIT={{ MEMORY_LIMIT }} -p MONGODB_USER={{ MONGODB_USER }} -p MONGODB_PASSWORD={{ MONGODB_PASSWORD }} -p MONGODB_DATABASE={{ MONGODB_DATABASE }} -p VOLUME_CAPACITY={{ VOLUME_CAPACITY }} -p MONGODB_VERSION={{ MONGODB_VERSION }} -p STORAGE_CLASS_NAME={{ STORAGE_CLASS_NAME }} | oc create --namespace={{ test_project_name }} -f -
+    tags:
+      - setup
+
+  - block:
+      - name: wait until the mongodb pod is running
+        shell: oc get pod -n "{{ test_project_name }}" | grep -v deploy | grep mongodb | grep Running
+        register: pod_ready_result
+        until: pod_ready_result.rc == 0
+        retries: 11
+        delay: 10
+        ignore_errors: True
+      - name: fail the test if the pod is NOT running after 120s
+        fail:
+          msg: "the pod is NOT running after 120s"
+        when: pod_ready_result|failed
+    tags:
+      - setup
+
+  - name: create ycsb
+    shell: oc create -f {{ tmp_folder }}/files/oc/dc_ycsb.yaml
+    tags:
+      - setup
+
+  - block:
+      - name: wait until the ycsb pod is running
+        shell: oc get pod -n "{{ test_project_name }}" | grep -v deploy | grep ycsb | grep Running
+        register: pod_ready_result
+        until: pod_ready_result.rc == 0
+        retries: 11
+        delay: 10
+        ignore_errors: True
+      - name: fail the test if the pod is NOT running after 120s
+        fail:
+          msg: "the pod is NOT running after 120s"
+        when: pod_ready_result|failed
+    tags:
+      - setup
+
+  - name: register pbench
+    shell: |
+      pbench-kill-tools
+      pbench-clear-tools
+      pbench-clear-results
+      pbench-register-tool-set
+      pbench-register-tool --name=oc
+
+      NODES=($(oc get node  --no-headers | grep -v master | awk '{print $1}'))
+
+      for node in "${NODES[@]}"
+      do
+        pbench-register-tool-set --remote="${node}"
+      done
+    when: pbench_registration|bool == true
+    tags:
+      - run
+
+  - block:
+    - name: create temporary file
+      tempfile:
+        state: file
+        suffix: temp
+      register: tmp_file_result
+    - name: show temporary file path
+      debug:
+        msg: "temporary file path: {{ tmp_file_result.path }}"
+    - name: run the benchmarks
+      command: pbench-user-benchmark --config="mongodb_storage_test_{{ STORAGE_CLASS_NAME }}" -- "{{ tmp_folder }}/files/scripts/test-mongo.sh" "{{ test_project_name }}" "{{ iteration }}" "{{ ycsb_threads }}" "{{ tmp_file_result.path }}"
+    tags:
+      - run
+
+  - block:
```

Comment:
this block is failing 
``` 
TASK [generate thoughtput file for pbench] ***************************************************************************************************************************************************
fatal: [ip-172-31-8-227.us-west-2.compute.internal]: FAILED! => {"changed": true, "cmd": "while read line;\n do\n grep Throughput \"${line}/result.txt\" > \"${line}/throughput_result.txt\"\n done < \"/tmp/ansible.WNkrlHtemp\"", "delta": "0:00:00.005157", "end": "2018-05-04 11:00:28.096849", "msg": "non-zero return code", "rc": 1, "start": "2018-05-04 11:00:28.091692", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
	to retry, use: --limit @/root/svt_hongkai/storage/mongodb/mongodb-test.retry
```

Reply:
Need to debug this with you.
My guess would be that the wanted line is not there for some reason.